### PR TITLE
Remove HasFlag calls

### PR DESCRIPTION
**Pull request by shargon**
Close https://github.com/neo-project/neo-vm/issues/188

### CODE REVIEW

File: src/neo-vm/Debugger.cs [hunk 1]
```diff
@@ -71,15 +71,15 @@ public VMState StepOut()
 
         public VMState StepOver()
         {
-            if (engine.State.HasFlag(VMState.HALT) || engine.State.HasFlag(VMState.FAULT))
+            if (engine.State == VMState.HALT || engine.State == VMState.FAULT)
                 return engine.State;
             engine.State &= ~VMState.BREAK;
```
Comment (shargon) [inline src/neo-vm/Debugger.cs]: I think that should be faster and cleaner change this sentences
```suggestion
            engine.State = VMState.NONE;
```

File: src/neo-vm/Debugger.cs [hunk 2]
```diff
@@ -62,7 +62,7 @@ public VMState StepOut()
         {
             engine.State &= ~VMState.BREAK;
```
Comment (shargon) [inline src/neo-vm/Debugger.cs]:
```suggestion
             engine.State = VMState.NONE;
```

File: src/neo-vm/Debugger.cs [hunk 3]
```diff
@@ -25,7 +25,7 @@ public void AddBreakPoint(Script script, uint position)
         public VMState Execute()
         {
             engine.State &= ~VMState.BREAK;
```
Comment (shargon) [inline src/neo-vm/Debugger.cs]:
```suggestion
             engine.State = VMState.NONE;
```

**Comment by devhawk** [comment]
Can we make this change in master-2.x as well?

**Comment by shargon** [comment]
> Can we make this change in master-2.x as well?

It's required for your tool?

**Comment by shargon** [comment]
Any update @neo-project/core?

**Comment by igormcoelho** [comment]
I don't understand these changes... and I don't understand how they related to the opened issue...

**Comment by shargon** [comment]
Ready to merge for me

File: src/neo-vm/ExecutionEngine.cs [hunk 4]
```diff
@@ -177,8 +177,9 @@ public virtual void Dispose()
 
         public VMState Execute()
         {
-            State &= ~VMState.BREAK;
-            while (!State.HasFlag(VMState.HALT) && !State.HasFlag(VMState.FAULT))
+            if (State == VMState.BREAK)
+                State = VMState.NONE;
+            while (State != VMState.HALT && State != VMState.FAULT)
```
Comment (igormcoelho) [inline src/neo-vm/ExecutionEngine.cs]: Ok, I got this part now.

File: src/neo-vm/Debugger.cs [hunk 5]
```diff
@@ -71,15 +73,15 @@ public VMState StepOut()
 
         public VMState StepOver()
         {
-            if (engine.State.HasFlag(VMState.HALT) || engine.State.HasFlag(VMState.FAULT))
+            if (engine.State == VMState.HALT || engine.State == VMState.FAULT)
                 return engine.State;
-            engine.State &= ~VMState.BREAK;
+            engine.State = VMState.NONE;
```
Comment (igormcoelho) [inline src/neo-vm/Debugger.cs]: In the part below, this was translated to:
```cs
            if (State == VMState.BREAK)
                State = VMState.NONE;
```

File: src/neo-vm/Debugger.cs [hunk 6]
```diff
@@ -71,15 +73,15 @@ public VMState StepOut()
 
         public VMState StepOver()
         {
-            if (engine.State.HasFlag(VMState.HALT) || engine.State.HasFlag(VMState.FAULT))
+            if (engine.State == VMState.HALT || engine.State == VMState.FAULT)
```
Comment (igormcoelho) [inline src/neo-vm/Debugger.cs]: Ok.

File: src/neo-vm/Debugger.cs [hunk 7]
```diff
@@ -50,7 +51,7 @@ public bool RemoveBreakPoint(Script script, uint position)
 
         public VMState StepInto()
         {
-            if (engine.State.HasFlag(VMState.HALT) || engine.State.HasFlag(VMState.FAULT))
+            if (engine.State == VMState.HALT || engine.State == VMState.FAULT)
```
Comment (igormcoelho) [inline src/neo-vm/Debugger.cs]: Ok.

File: src/neo-vm/Debugger.cs (hunk 5 above)
Comment (shargon) [reply src/neo-vm/Debugger.cs]: The previous condition check the other states

**Pull request merged**